balancer-js/src/modules/pools/pool-types/concerns/composableStable/join.concern.integration.spec.ts

```diff
@@ -22,146 +22,73 @@ const blockNumber = 16350000;
 const testPoolId =
   '0xa13a9247ea42d743238089903570127dda72fe4400000000000000000000035d';
 
-describe('ComposableStable pool join functions', async () => {
+describe('ComposableStable Pool - Join Functions', async () => {
   let signerAddress: string;
   let pool: PoolWithMethods;
-  // We have to rest the fork between each test as pool value changes after tx is submitted
-  beforeEach(async () => {
+  let testPoolHelper: TestPoolHelper;
+
+  before(async () => {
     signerAddress = await signer.getAddress();
 
-    const testPool = new TestPoolHelper(
+    testPoolHelper = new TestPoolHelper(
```

Comment:
Moved `testPoolHelper` instantiation to `before` instead of `beforeEach` so we avoid calling subgraph unnecessarily multiple times.